node/src/reactor/main_reactor/keep_up.rs

```diff
@@ -296,7 +296,7 @@ impl MainReactor {
                     // be required anymore
                     debug!("KeepUp: synced to TTL or Genesis");
                     self.block_synchronizer.purge_historical();
-                    self.sync_leaper.purge_sync_back_activity();
+                    self.sync_leaper.purge();
```

Comment:
This is probably not needed anymore since the purge now happens on state transition and it would catch this case as well. But I guess it won't hurt either since `KeepUp` only issues historical leaps so it won't affect the forward flow either.